hyperspy/events.py

```diff
@@ -54,16 +103,51 @@ def suppress(self):
             obj.val_b = b
         obj.events.values_changed.trigger()
         """
-        return EventSuppressionContext(self)
+        return EventsSuppressionContext(self)
 
 
 class Event(object):
 
     def __init__(self):
         self._connected = {}
-        self.suppress = False
+        self._suppress = False
+
+    def suppress(self, function=None):
```

Comment:
I agree that turned it into a function is better and more consistent. As a function, I bet that `suppress` is better than `suppressed`. I still think that it'll be better to split `suppress` in two: `suppress` and `suppress_callback`. As it is, it is harder to understand what it does without reading the docstring, while after splitting it it should be straightforward.